src/types/instantsearch.ts

```diff
@@ -4,11 +4,22 @@ import {
   SearchParameters,
   PlainSearchParameters,
 } from 'algoliasearch-helper';
-import { Index } from '../widgets/index/index';
 import { InsightsClient as AlgoliaInsightsClient } from './insights';
 import { Widget, UiState } from './widget';
-
-export type InstantSearchOptions = any;
+export { default as InstantSearch } from '../lib/InstantSearch';
+
+export type SearchClient = Pick<Client, 'search' | 'searchForFacetValues'>;
+
+export type InstantSearchOptions = {
+  indexName: string;
+  searchClient: SearchClient | Client;
+  numberLocale?: string;
+  searchFunction?: (helper: AlgoliaSearchHelper) => void;
+  searchParameters?: PlainSearchParameters;
+  routing?: any;
```

Comment:
Any reasons for not typing `routing`?

Reply:
I did not find a good existing type for it